### ISSUE TITLE
Feature request: Direct P2P connection using Hotspot 

### ISSUE THREAD

**Issue by Fearmipan** [localsend/localsend#2144]
### Is your feature request related to a problem? Please describe.

Useful if no AP/Router for sharing with friends in public/outdoor. 

### Describe the solution you'd like

Just use Hotspot for receiving the file.
Then the sender connect to Hotspot and send the file. 

### Describe alternatives you've considered

1. Add button "Generate QR" in receive page mwnu, then generate QR  with syntax...

`WIFI:S:<SSID Name>;T:WPS|WPA;P:<password>;H:true|false;;`
H = Hidden network

2. In sender page menu, add  scan QR code button to proceed Direct P2P Hotspot file transfer. 

### Additional context

I know it will be very complicated to do (require permission API, etc).
But it's very useful feature, and will speed up transfer rate, because no other client in the Hotspot

**Comment by JustAdreamerFL** [comment]
I think that this could work on android but not on the ios because of the ios limitations.

**Comment by Fearmipan** [comment]
> I think that this could work on android but not on the ios because of the ios limitations.

I see, iOS app not allowed to auto connect Wi-Fi network.

**Comment by ArtexJay** [comment]
Can just have this on android and pc then and tell the user on IOS that Apple doesnt allow for this feature to work properly.

**Comment by Fearmipan** [comment]
> Maybe Wi-Fi direct could be utilised for this?

AFAIK Wi-Fi Direct doesn't use HTTPS, is P2P protocol used to mirror display, transfer media, DLNA etc
LocalSend use HTTPS protocol to send any file.

**Comment by blaueente** [comment]
> > Maybe Wi-Fi direct could be utilised for this?
> 
> AFAIK, CMIIW. Wi-Fi Direct doesn't use HTTPS, is P2P protocol used to mirror display, transfer media, DLNA etc LocalSend use HTTPS protocol to send any file.

Yes, Wi-Fi Direct can absolutely be used. It works on a lower level. The android API will provide a channel where the application protocol can be normally used, such as the  localsend HTTPS protocol.
see https://developer.android.com/develop/connectivity/wifi/wifip2p

**Comment by Yiannis128** [comment]
On ios instructions on starting a hotspot could be provided instead so the user can do it manually.

Android/Linux/etc. should fully support it however!
